experimental/src/main/java/org/jboss/bacon/experimental/DependencyGenerator.java

```diff
@@ -18,85 +18,84 @@
-
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.Stack;
 
 @CommandLine.Command(
         name = "dependency-generator",
         description = "Automated dependency Build Config generator",
-        subcommands = { DependencyGenerator.Generate.class })
+        subcommands = { DependencyGenerator.Generate.class, DependencyGenerator.DepTree.class })
 @Slf4j
 public class DependencyGenerator {
 
-    @CommandLine.Command(name = "generate", description = "Generates build config")
-    public static class Generate extends ExperimentalCommand {
-
-        @CommandLine.Option(names = { "--project-dir" }, description = "Project directory")
-        private Path projectDir;
-
-        @CommandLine.Parameters(description = "Autobuilder configuration file")
-        private Path config;
+    @CommandLine.Command(name = "dependency-tree", description = "Outputs the analyzed dependency tree")
+    public static class DepTree extends DependencyGeneratorCommand {
 
         @Override
         public void run() {
-            try {
-                ObjectHelper.print(getJsonOutput(), generatePigConfig());
-            } catch (JsonProcessingException e) {
-                throw new FatalException("Caught exception " + e.getMessage(), e);
+            GeneratorConfig config = loadConfig();
+            DependencyResolver dependencyResolver = new DependencyResolver(config.getDependencyResolutionConfig());
+            DependencyResult dependencies;
+            if (projectDir != null) {
+                dependencies = dependencyResolver.resolve(projectDir);
+            } else {
+                dependencies = dependencyResolver.resolve();
             }
+            print(dependencies);
         }
 
-        private GeneratorConfig loadConfig() {
-            if (config == null) {
-                throw new FatalException("You need to specify the configuration directory!");
+        private void print(DependencyResult dependencies) {
+            Set<Project> entered = new HashSet<>();
+            Stack<StackElem> stack = new Stack<>();
+            for (Project topLevelProject : dependencies.getTopLevelProjects()) {
+                stack.push(new StackElem(topLevelProject, "# "));
             }
-
-            Yaml yaml = new Yaml(new Constructor(GeneratorConfig.class));
-            try (BufferedReader reader = Files.newBufferedReader(config)) {
-                GeneratorConfig config = yaml.load(reader);
-                validate(config);
-                return config;
-            } catch (IOException e) {
-                throw new FatalException("Unable to load config file", e);
+            while (!stack.isEmpty()) {
+                StackElem stackElem = stack.pop();
+                System.out.println(stackElem.prefix + stackElem.project.getFirstGAV());
```

Comment:
You want to log this or was it just for development?

Reply:
This should be the output of the command